drivers/clock_control/clock_control_silabs_siwx91x.c

```diff
@@ -202,6 +220,24 @@ static int siwx91x_clock_init(const struct device *dev)
 	siwx91x_clock_on(dev, (clock_control_subsys_t)SIWX91X_CLK_I2C1);
 #endif
 
+#ifdef CONFIG_PM
+	/* Initialize platform-specific configurations. */
+	sli_si91x_platform_init();
```

Comment:
Since this seems to have more to do with power manager than clocks, is there a good reason to have it in the clock driver instead of `soc.c` using one of the init hooks?